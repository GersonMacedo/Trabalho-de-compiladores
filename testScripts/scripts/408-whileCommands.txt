! -s RUN --disable-logs printer
! should correctly execute the while commands
program p ;
    var b: Boolean;
    var x: Integer;
begin
    print 0;
    while false do
        print 1
    ;
    print 2;
    b := true;
    while b do
    begin
        print 3;
        b := false;
    end ;
    print 4;
    x := 5;
    while x < 10 do
    begin
        print x;
        x := x + 1;
    end ;
    print 20;
end .