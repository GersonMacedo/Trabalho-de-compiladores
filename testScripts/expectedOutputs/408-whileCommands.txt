
Performing build generation and running at testScripts/scripts/408-whileCommands.txt
No errors found
Building the program

      PUSH      2
      LOADL     0
      CALL      putint
      CALL      puteol
      JUMP      L002
L001: LOADL     1
      CALL      putint
      CALL      puteol
L002: LOADL     0
      JUMPIF(1) L001
      LOADL     2
      CALL      putint
      CALL      puteol
      LOADL     1
      STORE(1)  [0]SB      #b
      JUMP      L004
L003: LOADL     3
      CALL      putint
      CALL      puteol
      LOADL     0
      STORE(1)  [0]SB      #b
L004: LOAD(1)   [0]SB      #b
      JUMPIF(1) L003
      LOADL     4
      CALL      putint
      CALL      puteol
      LOADL     5
      STORE(1)  [1]SB      #x
      JUMP      L006
L005: LOAD(1)   [1]SB      #x
      CALL      putint
      CALL      puteol
      LOAD(1)   [1]SB      #x
      LOADL     1
      CALL      add
      STORE(1)  [1]SB      #x
L006: LOAD(1)   [1]SB      #x
      LOADL     10
      CALL      lt
      JUMPIF(1) L005
      LOADL     20
      CALL      putint
      CALL      puteol
      POP(0)    2
      HALT
Builder::Writing object to obj.tam
********** TAM Interpreter (Java Version 2.1) **********
0
2
3
4
5
6
7
8
9
20

Program has halted normally.
